pkg/network/deviceinfo/deviceinfo_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+* This file is part of the KubeVirt project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Copyright the KubeVirt Authors.
+*
+ */
+package deviceinfo_test
+
+import (
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/network/deviceinfo"
+)
+
+var _ = Describe("DeviceInfo", func() {
+
+	const (
+		booHashedIfaceName  = "pod6446d58d6df"
+		fooHashedIfaceName  = "pod2c26b46b68f"
+		deviceInfoPlugin    = "deviceinfo"
+		nonDeviceInfoPlugin = "non_deviceinfo"
+	)
+
+	networkStatusWithMixedNetworksFmt := `[
+		{
+		    "name": "kindnet",
+		    "interface": "eth0",
+		    "ips": [
+		      "10.244.1.9"
+		    ],
+		    "mac": "3a:7e:42:fa:37:c6",
+            "default": true,
+		    "dns": {}
+	    },
+	    {
+		    "name": "default/nad1",
+		    "interface": "%s",
+		    "mac": "8a:37:d9:e7:0f:18",
+		    "dns": {}
+	    },
+	    {
+		    "name": "default/nad2",
+		    "interface": "%s",
+		    "dns": {},
+		    "device-info": {
+		      "type": "pci",
+		      "version": "1.0.0",
+		      "pci": {
+			    "pci-address": "0000:65:00.2"
+		      }
+		    }
+	    }
+    ]`
+
+	networkStatusWithPrimaryInterfaceOnly := `[
+	    {
+		    "name": "kindnet",
+		    "interface": "eth0",
+		    "ips": [
+		      "10.244.2.131"
+		    ],
+		    "mac": "82:cf:7c:98:43:7e",
+		    "default": true,
+		    "dns": {}
+	    }
+    ]`
+
+	bindingPlugins := map[string]virtv1.InterfaceBindingPlugin{
+		deviceInfoPlugin:    {DownwardAPI: virtv1.DeviceInfo},
+		nonDeviceInfoPlugin: {},
+	}
+
+	DescribeTable("should return an error",
+		func(networkStatusAnnotationValue string) {
+			networks := []virtv1.Network{*libvmi.MultusNetwork("foo", "default/nad1")}
+			interfaces := []virtv1.Interface{newBindingPluginInterface("foo", deviceInfoPlugin)}
+			_, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).To(HaveOccurred())
```

Comment:
```go
Expect(deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(
    networks,
    interfaces,
    networkStatusAnnotationValue,
    bindingPlugins,
)).NotTo(Succeed())
```

Reply:
it is not working.

Reply:
Ahh, my mistake, sorry.
It works only if the return value is the error (without data).